src/qml/editorwidgets/ValueMap.qml

```diff
@@ -77,28 +84,235 @@ EditorWidgetBase {
         verticalAlignment: Text.AlignVCenter
         horizontalAlignment: Text.AlignLeft
         elide: Text.ElideRight
-    }
+      }
 
-    background: Item {
-      implicitWidth: 120
-      implicitHeight: 36
+      background: Item {
+        implicitWidth: 120
+        implicitHeight: 36
 
-      Rectangle {
-        visible: !enabled
-        y: comboBox.height - 12
-        width: comboBox.width
-        height: comboBox.activeFocus ? 2 : 1
-        color: comboBox.activeFocus ? Theme.accentColor : Theme.accentLightColor
+        Rectangle {
+          visible: !enabled
+          y: comboBox.height - 12
+          width: comboBox.width
+          height: comboBox.activeFocus ? 2 : 1
+          color: comboBox.activeFocus ? Theme.accentColor : Theme.accentLightColor
+        }
+
+        Rectangle {
+          visible: enabled
+          anchors.fill: parent
+          id: backgroundRect
+          border.color: comboBox.pressed ? Theme.accentColor : Theme.accentLightColor
+          border.width: comboBox.visualFocus ? 2 : 1
+          color: Theme.controlBackgroundAlternateColor
+          radius: 2
+        }
       }
+    }
+
+    FontMetrics {
+        id: fontMetrics
+        font: comboBox.font
+    }
+
+    QfToolButton {
+        id: searchButton
+
+        Layout.preferredWidth: enabled ? 48 : 0
+        Layout.preferredHeight: 48
+
+        bgcolor: "transparent"
+        iconSource: Theme.getThemeIcon("ic_baseline_search_black")
+        iconColor: Theme.mainTextColor
 
-      Rectangle {
         visible: enabled
+
+        onClicked: {
+            searchFeaturePopup.open()
+        }
+    }
+
+    Popup {
```

Comment:
There's a lot of code duplication between that and the relation combobox. Could we try and generalize that into a reusable item for both the RelationCombobox and this value map editor widget?

Reply:
Yes I'll look how far is feasible. I was scared about breaking something

Reply:
@domi4484 , shall we merge this PR now, and you can try to de-duplication code in a follow up PR?

Reply:
@domi4484 , oh actually it'll need a rebase. PageHeader is no more, say hello to QfPageHeader :)

Reply:
yes lets merge it

Reply:
@nirvn Ill rebase!

Reply:
@nirvn done